include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -187,3 +188,25 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
 }
 };  // namespace onnxruntime
 #endif
+
+namespace onnxruntime {
+
+struct OrtLockFreeMutex {
```

Comment:
nit on naming: how about `OrtSpinLock`? calling a mutex lock-free seems odd to me.

Reply:
Good point.